### PR TITLE
feat: user table and modified auth routes

### PR DESCRIPTION
**Pull request by hanlinc27**
## Notion ticket link

[Create User Table](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* Update existing user table with phone number
* Edit auth route and services with updated types

## Steps to test
1. Drop the existing user table in PGAdmin
2. Run `docker-compose up --build`
3. See new users table

Screen Shot 2021-10-02 at 2 35 59 PM

## What should reviewers focus on?
* See if you can hit the new endpoint to create a user
Screen Shot 2021-10-02 at 2 36 11 PM

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/models/user.model.ts [hunk 1]
```diff
@@ -12,6 +13,9 @@ export default class User extends Model {
   @Column({ type: DataType.STRING })
   auth_id!: string;
 
-  @Column({ type: DataType.ENUM("User", "Admin") })
+  @Column({ type: DataType.ENUM("User", "Admin", "Volunteer", "Donor") })
   role!: Role;
+
+  @Column({ type: DataTypes.STRING })
+  phone_number!: string;
 }
```
Comment (Shehryar21) [inline backend/typescript/models/user.model.ts:21]: I believe email should also be added to the user model. Changes would need to be made in few other places as well to accommodate for this change.

**Review by Shehryar21 — CHANGES_REQUESTED** [review]
Once email field is added, I will review it again.

File: backend/typescript/models/user.model.ts (hunk 1 above)
Comment (hanlinc27) [reply backend/typescript/models/user.model.ts:21]: discussed, already referenced in `getUser()` service

**Review by Shehryar21 — APPROVED** [review]
LGTM!!

**Pull request merged**